panther_manager/src/plugins/action/call_set_bool_service_node.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_manager/plugins/action/call_set_bool_service_node.hpp>
+namespace panther_manager
+{
+
+bool CallSetBoolService::setRequest(typename Request::SharedPtr & request)
+{
+  if (!getInput<bool>("data", request->data)) {
+    RCLCPP_ERROR_STREAM(node_->get_logger(), "Failed to get input [data]");
```

Comment:
In ROS 1 it was throwing the exception because there was no other option to signalize the error. This function was `void`. Now you can return `false` and the tree will be in the  `FAILED` status rather than throw the exception.
 
 I think checking if tree is FAILED is more clear than catching the exception but if you prefer the exceptions we can restore the exceptions.

Reply:
Agree, it make sense. Working with the main tree I came to a conclusion that tree nodes should not throw any errors at runtime the exception to that may be throws inside constructor.